Source/WTF/wtf/Vector.h

```diff
@@ -56,269 +56,152 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(VectorBuffer);
+        else {
+            while (src != srcEnd) {
+                new (NotNull, dst) T(WTFMove(*src));
+                src->~T();
+                ++dst;
+                ++src;
+            }
+        }
     }
 
     static void move(std::span<T> src, std::span<T> dst)
     {
-        VectorMover<VectorTraits<T>::canMoveWithMemcpy, T>::move(std::to_address(src.begin()), std::to_address(src.end()), std::to_address(dst.begin()));
+        move(std::to_address(src.begin()), std::to_address(src.end()), std::to_address(dst.begin()));
     }
 
     static void moveOverlapping(T* src, T* srcEnd, T* dst)
     {
-        VectorMover<VectorTraits<T>::canMoveWithMemcpy, T>::moveOverlapping(src, srcEnd, dst);
+        if constexpr (VectorTraits<T>::canMoveWithMemcpy)
+            memmove(static_cast<void*>(dst), static_cast<void*>(const_cast<T*>(src)), reinterpret_cast<const char*>(srcEnd) - reinterpret_cast<const char*>(src));
+        else {
+            if (src > dst)
+                move(src, srcEnd, dst);
+            else {
+                T* dstEnd = dst + (srcEnd - src);
+                while (src != srcEnd) {
+                    --srcEnd;
+                    --dstEnd;
+                    new (NotNull, dstEnd) T(WTFMove(*srcEnd));
+                    srcEnd->~T();
+                }
+            }
+        }
     }
 
     static void moveOverlapping(std::span<T> src, std::span<T> dst)
     {
-        VectorMover<VectorTraits<T>::canMoveWithMemcpy, T>::moveOverlapping(std::to_address(src.begin()), std::to_address(src.end()), std::to_address(dst.begin()));
+        moveOverlapping(std::to_address(src.begin()), std::to_address(src.end()), std::to_address(dst.begin()));
     }
 
     static void uninitializedCopy(const T* src, const T* srcEnd, T* dst)
     {
-        VectorCopier<VectorTraits<T>::canCopyWithMemcpy, T>::uninitializedCopy(src, srcEnd, dst);
+        if constexpr (VectorTraits<T>::canCopyWithMemcpy)
+            memcpy(static_cast<void*>(dst), static_cast<void*>(const_cast<T*>(src)), reinterpret_cast<const char*>(srcEnd) - reinterpret_cast<const char*>(src));
+        else {
+            while (src != srcEnd) {
+                new (NotNull, dst) T(*src);
+                ++dst;
+                ++src;
+            }
+        }
     }
 
     static void uninitializedCopy(std::span<const T> src, std::span<T> dst)
     {
-        VectorCopier<VectorTraits<T>::canCopyWithMemcpy, T>::uninitializedCopy(std::to_address(src.begin()), std::to_address(src.end()), std::to_address(dst.begin()));
+        uninitializedCopy(std::to_address(src.begin()), std::to_address(src.end()), std::to_address(dst.begin()));
     }
 
     static void uninitializedFill(T* dst, T* dstEnd, const T& val)
     {
-        VectorFiller<VectorTraits<T>::canFillWithMemset, T>::uninitializedFill(dst, dstEnd, val);
+        if constexpr (VectorTraits<T>::canFillWithMemset) {
+            static_assert(sizeof(T) == 1);
+            memset(dst, val, dstEnd - dst);
```

Comment:
Seems like this will do something strange if `val` isn’t a scalar that converts to `int` without loss. I can imagine single bytes that when you convert them to `int` lose something. I’m surprised with all the other `static_assert` that we aren’t doing some assertions about that here. I guess it’s nothing really to worry about.